frontend/components/bc/BcContentFilter.vue

```diff
@@ -0,0 +1,107 @@
+<script lang="ts" setup>
+import { faMagnifyingGlass } from '@fortawesome/pro-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import type InputText from 'primevue/inputtext'
+
+interface Props {
+  searchPlaceholder?: string;
+  disabledFilter?: boolean;
+}
+const props = defineProps<Props>()
+
+defineEmits<{(e: 'filter-changed', value: string): void }>()
+
+const isFilterVisible = ref(false)
+const filter = ref('')
+
+const button = ref<{$el: HTMLButtonElement} | null>(null)
+
+const input = ref<{$el: HTMLInputElement} | null>(null)
+const focusAndSelect = (inputElement: {$el: HTMLInputElement}) => {
+  if (inputElement?.$el) {
+    // make sure the input is not disabled anymore
+    nextTick(() => {
+      inputElement.$el.focus()
+      inputElement.$el.select()
+    })
+  }
+}
+const closeFilter = () => {
+  isFilterVisible.value = false
+  button.value?.$el.focus()
+}
+const handleClick = () => {
+  isFilterVisible.value = !isFilterVisible.value
+  if (input.value) {
+    focusAndSelect(input.value)
+  }
+}
+</script>
+
+<template>
+  <div class="filter_elements_container">
+    <InputText
+      ref="input"
+      v-model="filter"
+      :placeholder="props.searchPlaceholder"
+      :disabled="!isFilterVisible"
+      :class="{ visible: isFilterVisible }"
+      @keydown.escape="closeFilter"
```

Comment:
add `stop`to prevent closing the dialog when this component is used in a dialog - like in the `ValidatorManagementModal.vue`

```suggestion
      @keydown.escape.stop="closeFilter"
```